bridge-adapters/kinetic-bridgehub-adapter-azure-rest/src/main/java/com/kineticdata/bridgehub/adapter/azure/rest/AzureRestApiHelper.java

```diff
@@ -139,6 +144,22 @@ public void getToken () throws BridgeError {
         }
     }
     
+    private void handleFailedReqeust (int responseCode) throws BridgeError {
```

Comment:
I think you should change this to accept the full HttpResponse object instead of just the code, then you can use the response entity in the BridgeError if you need to - like for 400 errors. You aren't handling 403, that is the Forbidden error - meaning the user is authenticated, but doesn't have the necessary permissions to make the request.

If all you are going to say is the status code and status reason phrase, you can rewrite this as:

```
private void handleFailedRequest(HttpResponse response) throws BridgeError {
    int statusCode = response.getStatusLine().getStatusCode();
    String reasonPhrase = response.getStatusLine().getReasonPhrase();
    throw new BridgeError(statusCode + ": " + reasonPhrase);
}
```

But it may make sense to do more than that, such as:

```
private void handleFailedRequest(HttpResponse response) throws BridgeError {
    int statusCode = response.getStatusLine().getStatusCode();
    String reasonPhrase = response.getStatusLine().getReasonPhrase();
    HttpEntity entity = response.getEntity();

    if (statusCode == 400) {
        throw new BridgeError("Bad request: " + EntityUtils.toString(entity));
    }
    else if (statusCode == 401) {
        LOGGER.debug("Not authorized, invalid authentication token");
        throw new BridgeError("Not authorized");
    }
    else {
        throw new BridgeError(statusCode + ": " + reasonPhrase);
    }
}
```